orchestrator/lambdas/layers/heimdall_utils/heimdall_utils/utils.py

```diff
@@ -96,18 +100,28 @@ def parse_timestamp(timestamp: Optional[str] = None) -> str:
     Notes:
         - The function uses the current system time to calculate the 3-month offset.
         - All returned timestamps are in UTC (denoted by the 'Z' suffix).
-        - The function logs a warning message when generating a default timestamp.
     """
-    format = "%Y-%m-%dT%H:%M:%SZ"
     try:
-        if timestamp and bool(datetime.strptime(timestamp, format)):
+        if timestamp and is_valid_timestamp(timestamp):
             return timestamp
     except (TypeError, ValueError):
         log.error("Timestamp is invalid")
 
-    log.warning("Generating Default timestamp")
+    log.debug("Generating Default timestamp")
+    default_timestamp = get_default_datetime()
+    result = default_timestamp.timestamp()
+
+    return time.strftime(TIMESTAMP_FORMAT, time.gmtime(result))
+
+
+def is_valid_timestamp(timestamp: str) -> bool:
+    default_timestamp = get_default_datetime()
+    current_timestamp = datetime.strptime(timestamp, TIMESTAMP_FORMAT)
+
+    return current_timestamp > default_timestamp
+
+
+def get_default_datetime() -> datetime:
     current_timestamp = datetime.now()
     three_months_ago = current_timestamp - timedelta(days=90)
-    result = three_months_ago.timestamp()
-
-    return time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(result))
+    return three_months_ago
```

Comment:
There's a lot implicit about this logic. Can we clear it up a little?

Parsing through the whole file, it seems like `is_valid_timestamp` is basically just testing if the given timestamp is more recent than 90 days ago, but just from reading `is_valid_timestamp` that's not immediately apparent. 

IMO, that's due to `get_default_datetime()` being a not very descriptive function name. Could we maybe refactor it to `get_90_days_ago_datetime` or even `get_datetime_from_past(days_ago)` (and then pass it `90`)